src/main/java/com/salesforce/storm/spout/dynamic/persistence/zookeeper/CuratorFactory.java

```diff
@@ -83,13 +85,21 @@ public static CuratorFramework createNewCuratorInstance(final Map<String, Object
             .collect(Collectors.joining(","));
 
         try {
+            // Create new ThreadFactory with named threads.
+            // TODO allow pushing in better naming.
+            final ThreadFactory threadFactory = new ThreadFactoryBuilder()
+                .setNameFormat("[DynamicSpout:PersistenceAdapter] Curator Pool %d")
```

Comment:
This is going to be confusing, if you're using the reference impl there are three uses of this with three distinct curator instances. Only one of them is actually the DS-PA.  I would say let's either add a String as a first param to this method or add a name to the config object.

Reply:
Going to leave this as is for now.  Its better than it was, but sounds like it's still far from ideal.